src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Triggers/DataBoxEdgeTriggerGetCmdletBase.cs

```diff
@@ -0,0 +1,150 @@
+using PSTopLevelResourceModel = Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models.PSDataBoxEdgeDevice;
+
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Common.Cmdlets.Triggers
+{
+    [Cmdlet(VerbsCommon.Get, Constants.Trigger, DefaultParameterSetName = ListParameterSet),
+     OutputType(typeof(PSResourceModel))]
+    public class DataBoxEdgeTriggerGetCmdletBase : AzureDataBoxEdgeCmdletBase
+    {
+        private const string ListParameterSet = "ListParameterSet";
+        private const string GetByNameParameterSet = "GetByNameParameterSet";
+        private const string GetByResourceIdParameterSet = "GetByResourceIdParameterSet";
+        private const string GetByParentObjectParameterSet = "GetByParentObjectParameterSet";
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = GetByResourceIdParameterSet,
+            HelpMessage = Constants.ResourceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = ListParameterSet,
+            HelpMessage = Constants.ResourceGroupNameHelpMessage,
+            Position = 0)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = GetByNameParameterSet,
+            HelpMessage = Constants.ResourceGroupNameHelpMessage,
+            Position = 0)]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = ListParameterSet,
+            HelpMessage = Constants.DeviceNameHelpMessage,
+            Position = 1)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = GetByNameParameterSet,
+            HelpMessage = Constants.DeviceNameHelpMessage,
+            Position = 1)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter("Microsoft.DataBoxEdge/dataBoxEdgeDevices", nameof(ResourceGroupName))]
+        public string DeviceName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = GetByNameParameterSet,
+            HelpMessage = Constants.NameHelpMessage,
+            Position = 2)]
+        [Parameter(Mandatory = false,
+            ParameterSetName = GetByParentObjectParameterSet,
+            HelpMessage = HelpMessageTrigger.NameHelpMessage
+        )]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true,
+            ParameterSetName = GetByParentObjectParameterSet,
+            HelpMessage = Constants.PsDeviceObjectHelpMessage)]
+        [ValidateNotNull]
+        public PSTopLevelResourceModel DeviceObject;
+
+        private ResourceModel GetResourceModel()
+        {
+            return this.DataBoxEdgeManagementClient.Triggers.Get(this.DeviceName, this.Name, this.ResourceGroupName);
+        }
+
+        private List<PSResourceModel> GetByResourceName()
```

Comment:
It's better to rename it as `ListResourceByResourceName`

Reply:
And this function name contains `ByResourceName` but the code doesn't contain `name`. It wil be more clearly:
```C#
private List<PSResourceModel> GetByResourceName()
{
    var resourceModel = this.DataBoxEdgeManagementClient.Triggers.Get(this.DeviceName, this.Name, this.ResourceGroupName);
    return new List<PSResourceModel>() {PSResourceModel.PSDataBoxEdgeTriggerObject(resourceModel)};
}
```
or
```C#
private List<PSResourceModel> GetByResourceName()
{
    var resourceModel = GetResourceModel(this.DeviceName, this.Name, this.ResourceGroupName);
    return new List<PSResourceModel>() {PSResourceModel.PSDataBoxEdgeTriggerObject(resourceModel)};
}
```

Reply:
Yeah, I think ListByName makes more sense. will update that

Reply:
Sure~